packages/roosterjs-editor-types/lib/event/BeforePasteEvent.ts

```diff
@@ -37,6 +38,11 @@ export interface BeforePasteEventData {
      * Attributes of the root "HTML" tag
      */
     htmlAttributes: Record<string, string>;
+
+    /**
+     * Paste type option (as plain text, merge format, normal, as image)
+     */
+    pasteType?: PasteType;

```

Comment:
Also need to allow `CompatiblePasteType`:

pasteType: PasteType | CompatiblePasteType;

Make sure to "import type" only for the Compatible type
And make this as required so that we know we can always read this value

Reply:
Also make it as readonly, so plugin cannot change it. Because this value comes from user's decision, plugins should never change it.